content/en/blog/_posts/2024-08-15-consistent-read-from-cache.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: blog
+title: 'Kubernetes v1.31: Accelerating Cluster Performance with Consistent Reads from Cache'
+date: 2024-08-15
+slug: consistent-read-from-cache-beta
+author: >
+  Marek Siarkowicz (Google)
+---
+
+Kubernetes is renowned for its robust orchestration of containerized applications,
+but as clusters grow, the demands on the control plane can become a bottleneck.
+A key challenge has been ensuring strongly consistent reads from the etcd datastore,
+requiring resource-intensive quorum reads.
+
+Today, the Kubernetes community is excited to announce a major improvement:
+**Consistent Reads from Cache**, graduating to Beta in Kubernetes 1.31.
+
+### Why consistent reads matter
+
+Consistent reads are essential for ensuring that Kubernetes components have an accurate view of the latest cluster state.
+Guaranteeing consistent reads is crucial for maintaining the accuracy and reliability of Kubernetes operations,
+enabling components to make informed decisions based on up-to-date information.
+In large-scale clusters, fetching and processing this data can be a performance bottleneck,
+especially for requests that involve filtering results.
+While Kubernetes can filter data by namespace directly within etcd,
+any other filtering by labels or field selectors requires the entire dataset to be fetched from etcd and then filtered in-memory by the Kubernetes API server.
+This is particularly impactful for components like the kubelet,
+which only needs to list pods scheduled on its node but forces the API Server and etcd to process all pods in the cluster.
```

Comment:
```suggestion
which only needs to list pods scheduled to its node - but previously required the API Server and etcd
to do a fresh fetch from etcd, then process every pod in the cluster.
```
?

Reply:
Feel free to reword. The key thing is to compare the behavior with something that is now in the past tense.

Reply:
Done